app/azure/Dashboard/DashboardView.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEnvironmentId } from '@/portainer/hooks/useEnvironmentId';
+import { PageHeader } from '@/portainer/components/PageHeader';
+import { Widget, WidgetBody } from '@/portainer/components/widget';
+import { r2a } from '@/react-tools/react2angular';
+
+import { useResourceGroups, useSubscriptions } from '../queries';
+
+export function DashboardView() {
+  const environmentId = useEnvironmentId();
+  const { data: subscriptions, isLoading: isLoadingSubscriptions } =
+    useSubscriptions(environmentId);
+  const { resourceGroups, isLoading: isLoadingResourceGroups } =
+    useResourceGroups(environmentId, subscriptions);
+  const isLoading = isLoadingSubscriptions || isLoadingResourceGroups;
+
+  // Inspired by azureService.aggregate
+  let aggregatedResources: unknown[] = [];
```

Comment:
I think resourceGroups[key] is of type ResourceGroup. use it instead of `unknown`

Reply:
but since you need the length, just do `Object.values(resourceGroups).length`

Reply:
Thats what I tried previously, however the useResourceGroups hook returns data nested inside each subscription id rather than aggregate data so it gives you the number of subscriptions so I needed to aggregate the resource groups data myself. Should I add this aggregating code into the util file so it can be reused?

Reply:
Also thanks for the heads up about the type, I am still getting my head around it 🧠